include/asm/warning.h

```diff
@@ -21,39 +21,48 @@ enum WarningState {
 };
 
 enum WarningID {
-	WARNING_ASSERT,		      /* Assertions */
-	WARNING_BACKWARDS_FOR,	      /* `for` loop with backwards range */
-	WARNING_BUILTIN_ARG,	      /* Invalid args to builtins */
-	WARNING_CHARMAP_REDEF,        /* Charmap entry re-definition */
-	WARNING_DIV,		      /* Division undefined behavior */
-	WARNING_EMPTY_DATA_DIRECTIVE, /* `db`, `dw` or `dl` directive without data in ROM */
-	WARNING_EMPTY_MACRO_ARG,      /* Empty macro argument */
-	WARNING_EMPTY_STRRPL,	      /* Empty second argument in `STRRPL` */
-	WARNING_LARGE_CONSTANT,	      /* Constants too large */
-	WARNING_LONG_STR,	      /* String too long for internal buffers */
-	WARNING_MACRO_SHIFT,	      /* Shift past available arguments in macro */
-	WARNING_NESTED_COMMENT,	      /* Comment-start delimiter in a block comment */
-	WARNING_OBSOLETE,	      /* Obsolete things */
-	WARNING_SHIFT,		      /* Shifting undefined behavior */
-	WARNING_SHIFT_AMOUNT,	      /* Strange shift amount */
-	WARNING_TRUNCATION,	      /* Implicit truncation loses some bits */
-	WARNING_USER,		      /* User warnings */
+	WARNING_ASSERT,		      // Assertions
+	WARNING_BACKWARDS_FOR,	      // `for` loop with backwards range
+	WARNING_BUILTIN_ARG,	      // Invalid args to builtins
+	WARNING_CHARMAP_REDEF,        // Charmap entry re-definition
+	WARNING_DIV,		      // Division undefined behavior
+	WARNING_EMPTY_DATA_DIRECTIVE, // `db`, `dw` or `dl` directive without data in ROM
+	WARNING_EMPTY_MACRO_ARG,      // Empty macro argument
+	WARNING_EMPTY_STRRPL,	      // Empty second argument in `STRRPL`
+	WARNING_LARGE_CONSTANT,	      // Constants too large
+	WARNING_LONG_STR,	      // String too long for internal buffers
+	WARNING_MACRO_SHIFT,	      // Shift past available arguments in macro
+	WARNING_NESTED_COMMENT,	      // Comment-start delimiter in a block comment
+	WARNING_OBSOLETE,	      // Obsolete things
+	WARNING_SHIFT,		      // Shifting undefined behavior
+	WARNING_SHIFT_AMOUNT,	      // Strange shift amount
+	WARNING_USER,		      // User warnings
 
-	NB_WARNINGS,
+	NB_PLAIN_WARNINGS,
+
+	// Warnings past this point are "parametric" warnings, only mapping to a single flag
+#define PARAM_WARNINGS_START NB_PLAIN_WARNINGS
+	// Implicit truncation loses some bits
+	WARNING_TRUNCATION_1 = PARAM_WARNINGS_START,
+	WARNING_TRUNCATION_2,
 
-	/* Warnings past this point are "meta" warnings */
-	WARNING_ALL = NB_WARNINGS,
+	NB_WARNINGS_PARAM,
```

Comment:
:pick: `NB_PLAIN_AND_PARAM_WARNINGS` would be more verbose but I think more clear. (No confusion with `NB_PARAM_WARNINGS`.)